### DISCUSSION TITLE
Install and run Minimed connect to NS on Google Cloud VM

### DISCUSSION THREAD

**Discussion by adipanait** [NightscoutFoundation/xDrip#2566] [Q&A] [answered]
Hi everyone. 

Many users including the undersigned are using Medtronic sensors (Guardian 3 or 4) & need to run an existing utility (https://github.com/FredMK/minimed-connect-to-nightscout-wrapper) on a server - usually on a VM or on a laptop or phone running the app in a terminal, at home. Personally, I run this app on al old Android phone running Termux (awesome terminal emulator for Android, with Wake lock).  

Since NS now works well on Google Cloud- I better option would be to be able to have both NS and "minimed -connect-to-nightscout" on the same VM. 

I'm not very familiar with how to work with Ubuntu VM's, so I need help figuring out:
- how to copy the needed folder/ files on the VM
- how to install Node.js
- what's needed (permissions, other settings) to install all dependencies & keep the program running on the VM so that it sends Medtronic sensor readings to NS. 

@Navid200 - as talked; 

Thank you!

Adrian

**Comment by Navid200** [comment]
Do you have a test machine?  You can create one to experiment with before committing.

I can then walk you through the steps required to follow what is explained as the required steps in that repository you have linked.
I have no experience with that particular app.  But, the steps are quite simple.
If it works in your test setup, you can repeat for your main setup.

**Reply by Navid200** [reply]
I am not sure how this works.  If it needs to access Nightscout through the ip address, it may not be that simple.

@tzachi-dar

**Reply by adipanait** [reply]
It uses the parameters set in .env, namely - 

MMCONNECT_SERVER=
CARELINK_USERNAME=
CARELINK_PASSWORD=
CARELINK_QUIET=true

# Nightscout API SECRET
API_SECRET=
# Nightscout HOSTNAME
WEBSITE_HOSTNAME=

If possible, I would test on the existing Google Cloud VM, nothing to break, right? If we do break something - I'll go ahead and create a new one just to keep having NS working (I like to avoid creating a new VM and using yet another single use Revolut credit card).

**Reply by Navid200** [reply]
Before we jump the gun, can we talk a little bit?
Do you know that xDrip now supports Carelink?

Can you tell me how your system works and why you need this?

**Comment by adipanait** [comment]
Sure, so:
1. I use Medtronic sensors AND Loop - FreeAPS version (iOS), with Omnipod as insulin pump and  Nightscout as Bg source
2. in order to get the sensor values to Nightscout - I need the utility mentioned above, that connects via Medtronic API to their cloud service, each 5 minutes (actually the read interval can be set).
3. The only way I found so far to make things work in this setup was with running the app in a terminal window on another device. For now it's an Android phone, it can be done from my Mac too, if needed. It is inconvenient because of the extra device, and I would not install and run it on the same iPhone I use for Loop since it will eat additional battery (Guardian connect is already a big eater). 

That's why it would be great to try and run it, in a terminal window, on the same Google Cloud VM where I have my Nightscout. 

Hope this helps

**Reply by Navid200** [reply]
I understand that the utility sends data to Nightscout.
What I don't understand is where it gets its input from.

**Reply by adipanait** [reply]
oh, it gets it by querying Carelink (https://carelink.medtronic.eu/), which is the equivalent of Dexcom Share. It has the login details in the .env file (as shared above).

**Reply by Navid200** [reply]
Ideally, we would find a way to just transfer the data directly to Nightscout on the virtual machine.  
Using the hostname to transfer data will add egress.  So, there will be added traffic.  But, the source and destination are on the same machine.

**Reply by Navid200** [reply]
I was hoping that @tzachi-dar would provide some tips.

As long as you have backed up your data, you can always delete and create a new machine if worse comes to worst.
Then, you can restore.
https://navid200.github.io/xDrip/docs/Nightscout/Reinstall.html

**Comment by Navid200** [comment]
@tzachi-dar Can a utility installed on the virtual machine access Nightscout using the hostname considering it is already on the virtual machine?

**Comment by adipanait** [comment]
I managed to create a folder using cloud shell on the VM (mkdir ...). Then, I copied the whole folder I had on my local machine  (making sure hidden files are included too, so that .env gets copied too). Then, i've started the utility with "node run.js" - and it works (node.js is preinstalled on the VM). 

The one thing I need help with is- how do I keep this shell running; I've read that it automatically disconnects after a max of 12 hours.

**Reply by Navid200** [reply]
How did you keep it going when you ran it on your old Android phone?

**Comment by adipanait** [comment]
Termux (the terminal emulator app) has this option - Acquire wakelock.

Adrian Panait
Digital Accessibility
________________________________

**Comment by Navid200** [comment]
@tzachi-dar Can @adipanait use screen in order to keep the linux code running continuously?

**Comment by adipanait** [comment, marked as answer]
I believe I've got it running on the VM, I am logged out of SSH and Nightscout keeps working. The key was using the nohup command (see last step below). 

**Steps:**
1. Open a SSH window. If presented with the Nightscout menu - exit to terminal;
2. install git: "sudo apt-get install git"
3. clone the repository on your VM: "git clone https://github.com/FredMK/minimed-connect-to-nightscout-wrapper"
4. move into that directory "cd minimed-connect-to-nightscout-wrapper" (of course by using arrow key and replacing everything before the directory name with cd)
5. type "npm install". This will install all needed dependencies -it will take some time, just leave it open until you're back to the regular command prompt. 
6. rename the example config file to .env by typing "mv .env.example .env"
7. edit the .env file with your variables, as instructed in FredMK's Github repo. I used: "nano .env". Ctrl+O to save, Ctrl+X to exit. 
8. type "nohup node run.js &" - to start the command in the background and keep it running after logout. 

I didn't use this yet- but, to stop the process for whatever reason: list all running processes "ps -ef", then "kill <needed ID>".

**Reply by Navid200** [reply]
git is already installed.

I am glad you found a solution.
